letsencrypt/storage.py

```diff
@@ -61,21 +62,27 @@ def write_renewal_config(filename, target, relevant_data):
     :rtype: configobj.ConfigObj
 
     """
-    # create_empty creates a new config file if filename does not exist
-    config = configobj.ConfigObj(filename, create_empty=True)
+    config = configobj.ConfigObj(o_filename)
     for kind in ALL_FOUR:
         config[kind] = target[kind]
 
-    if relevant_data:
-        config["renewalparams"] = relevant_data
+    if "renewalparams" not in config:
+        config["renewalparams"] = {}
         config.comments["renewalparams"] = ["",
                                             "Options used in "
                                             "the renewal process"]
 
+    config["renewalparams"].update(relevant_data)
+
+    for k in config["renewalparams"].keys():
+        if k not in relevant_data:
+            del config["renewalparams"][k]
```

Comment:
One thing I noticed is that by doing this after the line `config["renewalparams"].update(relevant_data)`, comments are preserved even if the value is modified. For example, if I have a renewal config file with this line:
```
authenticator = apache # my authenticator is apache
```
After running:
```
letsencrypt renew --force-renewal --webroot -w /var/www/html
```
The line becomes:
```
authenticator = webroot# my authenticator is apache
```
Is this the behavior we want? It might be, but I thought I'd ask the question.